packages/calcite-components/src/utils/interactive.ts

```diff
@@ -104,3 +121,76 @@ export function updateHostInteraction(
 
   component.el.removeAttribute("aria-disabled");
 }
+
+function blockInteraction(component: InteractiveComponent): void {
+  component.el.click = interceptedClick;
+  addInteractionListeners(isFirefox ? getParentElement(component) : component.el);
+}
+
+function addInteractionListeners(element: HTMLElement): void {
+  if (!element) {
+    // this path is only applicable to Firefox
+    return;
+  }
+
+  element.addEventListener("pointerdown", onPointerDown, captureOnlyOptions);
+  nonBubblingWhenDisabledMouseEvents.forEach((event) =>
+    element.addEventListener(event, onNonBubblingWhenDisabledMouseEvent, captureOnlyOptions)
+  );
+}
+
+function getParentElement(component: InteractiveComponent): ParentElement {
+  return interactiveElementToParent.get(component.el as InteractiveHTMLElement);
+}
+
+function restoreInteraction(component: InteractiveComponent): void {
+  delete component.el.click; // fallback on HTMLElement.prototype.click
+  removeInteractionListeners(isFirefox ? getParentElement(component) : component.el);
+}
+
+function removeInteractionListeners(element: HTMLElement): void {
+  if (!element) {
+    // this path is only applicable to Firefox
+    return;
+  }
+
+  element.removeEventListener("pointerdown", onPointerDown, captureOnlyOptions);
+  nonBubblingWhenDisabledMouseEvents.forEach((event) =>
+    element.removeEventListener(event, onNonBubblingWhenDisabledMouseEvent, captureOnlyOptions)
+  );
+}
+
+/**
+ * This utility helps disable components consistently in Firefox.
+ *
+ * It needs to be called in `connectedCallback` and is only needed for Firefox as it does not call capture event listeners before non-capture ones (see https://bugzilla.mozilla.org/show_bug.cgi?id=1731504).
+ *
+ * @param component
+ */
+export function connectInteractive(component: InteractiveComponent): void {
+  if (!component.disabled || !isFirefox) {
+    return;
+  }
+
+  const parent =
+    component.el.parentElement || component.el; /* assume element is host if it has no parent when connected */
```

Comment:
would `getShadowRootNode()` help here?

Reply:
It would help for sure, but I wanted to keep it simple since I only need to check for the parent or the host. Worth noting that most of these changes will be dropped once Firefox ESR is bumped to a version matching 116+. 🎉